src/api/procedures/modifyInstructionAffirmation.ts

```diff
@@ -175,16 +185,104 @@ export async function getAuthorization(
   };
 }
 
+/**
+ * @hidden
+ */
+function extractPortfolioParams(params: ModifyInstructionAffirmationParams): PortfolioLike[] {
+  const { operation } = params;
+  let portfolioParams: PortfolioLike[] = [];
+  if (operation === InstructionAffirmationOperation.Reject) {
+    const { portfolio } = params;
+    if (portfolio) {
+      portfolioParams.push(portfolio);
+    }
+  } else {
+    const { portfolios } = params;
+    if (portfolios) {
+      portfolioParams = [...portfolioParams, ...portfolios];
+    }
+  }
+  return portfolioParams;
+}
+
+/**
+ * @hidden
+ */
+const assemblePortfolios = async (
+  result: [(DefaultPortfolio | NumberedPortfolio)[], BigNumber],
+  from: DefaultPortfolio | NumberedPortfolio,
+  to: DefaultPortfolio | NumberedPortfolio,
+  signingDid: string,
+  portfolioIdParams: PortfolioId[]
+): Promise<[(DefaultPortfolio | NumberedPortfolio)[], BigNumber]> => {
+  const [fromExists, toExists] = await Promise.all([from.exists(), to.exists()]);
+
+  const [custodiedPortfolios, amount] = result;
+
+  let res = [...custodiedPortfolios];
+  let legAmount = amount;
+
+  const checkCustody = async (
+    legPortfolio: DefaultPortfolio | NumberedPortfolio,
+    exists: boolean,
+    sender: boolean
+  ): Promise<void> => {
+    if (exists) {
+      const isCustodied = await legPortfolio.isCustodiedBy({ identity: signingDid });
+      if (isCustodied) {
+        res = [...res, legPortfolio];
+        if (sender) {
+          legAmount = legAmount.plus(1);
+        }
+      }
+    } else if (legPortfolio.owner.did === signingDid) {
+      res = [...res, legPortfolio];
+    }
+  };
+
+  const isParam = (legPortfolio: DefaultPortfolio | NumberedPortfolio): boolean => {
+    const { did: legPortfolioDid, number: legPortfolioNumber } =
+      portfolioLikeToPortfolioId(legPortfolio);
+    return (
+      !portfolioIdParams.length ||
+      portfolioIdParams.some(
+        ({ did, number }) =>
+          did === legPortfolioDid &&
+          new BigNumber(legPortfolioNumber || 0).eq(new BigNumber(number || 0))
```

Comment:
I think here we don't need to convert to BigNumber to compare since these aren't exposed anywhere, just a logic check with something that should fit into normal JS numbers.

OK to leave